environs/networking.go

```diff
@@ -94,7 +94,7 @@ type Networking interface {
 	ReleaseContainerAddresses(ctx context.ProviderCallContext, interfaces []network.ProviderInterfaceInfo) error
 
 	// SSHAddresses filters provided addresses for addresses usable for SSH
-	SSHAddresses(ctx context.ProviderCallContext, addresses []corenetwork.Address) ([]corenetwork.Address, error)
+	SSHAddresses(ctx context.ProviderCallContext, addresses corenetwork.SpaceAddresses) (corenetwork.SpaceAddresses, error)
```

Comment:
This is a bit of an odd man out.

Usually we would have the provider only deal with `ProviderAddresses`. This is called from the sshclient API facade to filter machine addresses (which are `SpaceAddresses`). 

Each provider generally does one of two things:
- just returns all the addresses back or;
- returns a subset based on public scope matching.

The address `Value` is then returned to the client, which is just a string, so we do not actually leak a `SpaceAddress`.

Reply:
Done.

Reply:
Done.